src/environments/README.md

```diff
@@ -1,15 +1,138 @@
+
+Permanent environments are long-running environments that will be constantly updated and contributed to.
+These environments persist beyond the lifecycle of the process and will be cleaned up manually.
+
+You can create an environment using `arcctl create environment`. The example below creates an environment
+called "production" on a datacenter called "my-datacenter":
+
+```sh
+$ arcctl create environment production --datacenter my-datacenter
+```
+
+_Note: Creating even empty environments may require the creation of cloud resources
+depending on what your datacenter configuration requires._
+
+You can also create environments and immediately load them with contents by providing
+an environment configuration as an argument for the create command:
+
+```sh
+$ arcctl create environment production ./environment.yml --datacenter my-datacenter
+```
+
+## Updating environments
+
+### Using the deploy command
+
+The most common way for developers to provide targeted updates to existing environments is through
+the `arcctl deploy` command. This command allows developers to deploy a single component into an
+existing environment and pass in a few key configuration elements.
+
+```sh
+$ arcctl deploy architect/auth:latest --environment production
 ```
 
-## Registering environments
+One example of configuration that can be passed in the deploy command is ingress subdomains. Domain
+names in the same environment often use the same base URL any naming conventions for exposing
+ingress rules to the outside world, but that means that means that ingress names for components can
+collide with one another when they land in the same environment (e.g. two components with an ingress
+named "api").
+
+To resolve these conflicts, you can specify a subdomain manually with the `--ingress, -i` flag:
 
 ```sh
-$ arcctl environment apply ./environment.yml --name my-env --datacenter my-datacenter
+$ arcctl deploy architect/auth:latest --environment production --ingress api:auth-api frontend:auth
 ```
+
+The command above will assign the subdomains `auth-api` and `auth` to the `api` and `frontend` ingress
+rules respectively inside the `architect/auth` component.
+
+_The deploy command is basically a wrapper on top of the `arcctl environment update` command seen below.
+It mutates the underlying environment configuration and then re-applies the changes. If you don't see
+a flag to configure what you need, check out the environment configuration file details below._
+
+### From a configuration file
+
+Environments can be updated at any time using the `arcctl update environment` command. The update
+command will ensure the environment matches the provided configuration. That sometimes means creating
+new resources, destroying resources that are no longer needed, or both.
+
+```sh
+$ arcctl update environment production ./environment.yml
+```
+
+You can also return the environmnent to an empty state by simply NOT providing an environment configuration
```

Comment:
That does remind me that we definitely need a pre-flight "plan" equivalent so that the process doesn't just start. We should show them the diff table first and then prompt for approval.

Now tracked in https://github.com/architect-team/arcctl/issues/66